### DISCUSSION TITLE
How to configure the lock file path

### DISCUSSION THREAD

**Discussion by Xcliu** [eclipse-iceoryx/iceoryx#1664] [Q&A] [answered]
## Brief feature description
Are there some mechanisms which can be used to configure the lock file path (something like the toml file or callable API)?  

## Detailed information
NONE

**Comment by elfenpiff** [comment, marked as answer]
@Xcliu I do not know if you mean the lock file which roudi creates or the underlying `FileLock` construct in our posix wrappers so you get an answer for both :smile: 

1. **RouDi lock file path** This can be configured platform specific in the file `iceoryx_platform/YOUR_PLATFORM/include/iceoryx_platform/platform_settings.hpp` by adjusting the variable `IOX_LOCK_FILE_PATH_PREFIX`. 
Lets assume you are on our standard `linux` platform and would like to change the path to `/home/iceoryx/` you would open `iceoryx_platform/linux/include/iceoryx_platform/platform_settings.hpp` and change the line to `constexpr const char IOX_LOCK_FILE_PATH_PREFIX[] = "/home/iceoryx";`
The disadvantage is that this is only temporarily since the file is managed in our repo. To make this permanently you have to create your own platform - see hint below.


2. **Directory of posix::FileLock** When you would like to create a file lock manually you can use the `FileLockBuilder` and can use any existing directory. For instance:
```cxx
#include "iceoryx_hoofs/posix_wrapper/file_lock.hpp"

auto fileLock = iox::posix::FileLockBuilder().name("myLockName")
                                              .path("/home/iceoryx")
                                              .permission(iox::cxx::perms::owner_all)
                                              .create()
                                              .expect("Oh no I couldn't create the lock");
```

## Hint

If you would like to make this file lock path change permanently in your platform you can create easily your own platform with all your custom settings, see this article: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/custom-iceoryx-platform.md

Let's say your platform should be based on our `linux` platform then you can just copy the `iceoryx_platform/linux` directory, add all the changes you like and then use it like it is described in the article above.

## Future

At the moment we do not have this on our radar as a runtime feature so that you can provide this via command line or config file. The problem here is that roudi and every application either require the same command line parameters or have to read the identical config file which often leads to problems. When the configuration differs a lot can go wrong.
But a CMake argument would maybe a suitable alternative.

@Xcliu what would you prefer? Or are you fine with the custom platform approach?

**Comment by elfenpiff** [comment]
@Xcliu I will transform this issue into discussion which closes the issue for now. If some work follows out of the discussion we can create another issue.

**Comment by Xcliu** [comment]
Thank you so much!  This answer is perfect for me.
